Tools/LyTestTools/ly_test_tools/launchers/platforms/linux/launcher.py

```diff
@@ -0,0 +1,224 @@
+        # type: () -> None
+        """
+        Checks the returncode of the launcher if it exists. Raises a CrashError if the returncode is non-zero. Returns
+        None otherwise. This function must be called after exiting the launcher properly and NOT using its provided
+        teardown(). Provided teardown() will always return a non-zero returncode and should not be checked.
+
+        :return: None
+        """
+        return_code = self.get_returncode()
+        if return_code != 0:
+            log.error(f"Launcher exited with non-zero return code: {return_code}")
+            raise ly_test_tools.launchers.exceptions.CrashError()
+        return None
+
+    def configure_settings(self):
+        """
+        Configures system level settings and syncs the launcher to the targeted console IP.
+
+        :return: None
+        """
+        # Update settings via the settings registry to avoid modifying the bootstrap.cfg
+        host_ip = '127.0.0.1'
+        self.args.append(f'--regset="/Amazon/AzCore/Bootstrap/project_path={self.workspace.paths.project()}"')
+        self.args.append(f'--regset="/Amazon/AzCore/Bootstrap/remote_ip={host_ip}"')
+        self.args.append('--regset="/Amazon/AzCore/Bootstrap/wait_for_connect=1"')
+        self.args.append(f'--regset="/Amazon/AzCore/Bootstrap/allowed_list={host_ip}"')
+
+        self.workspace.settings.modify_platform_setting("r_AssetProcessorShaderCompiler", 1)
+        self.workspace.settings.modify_platform_setting("r_ShaderCompilerServer", host_ip)
+        self.workspace.settings.modify_platform_setting("log_RemoteConsoleAllowedAddresses", host_ip)
+
+
+class DedicatedLinuxLauncher(LinuxLauncher):
+
+    def setup(self, backupFiles=True, launch_ap=False):
+        """
+        Perform setup of this launcher, must be called before launching.
+        Subclasses should call its parent's setup() before calling its own code, unless it changes configuration files
+
+        :param backupFiles: Bool to backup setup files
+        :param lauch_ap: Bool to lauch the asset processor
+        :return: None
+        """
+        # Base setup defaults to None
+        if launch_ap is None:
+            launch_ap = False
+
+        super(DedicatedLinuxLauncher, self).setup(backupFiles, launch_ap)
+
+    def binary_path(self):
+        """
+        Return full path to the dedicated server launcher for the build directory.
+
+        :return: full path to <project>Launcher_Server
+        """
+        assert self.workspace.project is not None, (
+            'Project cannot be NoneType - please specify a project name string.')
+        return os.path.join(f"{self.workspace.paths.build_directory()}",
+                            f"{self.workspace.project}.ServerLauncher")
+
+
+class LinuxEditor(LinuxLauncher):
+
+    def __init__(self, build, args):
+        super(LinuxEditor, self).__init__(build, args)
+        self.args.append('--regset="/Amazon/Settings/EnableSourceControl=false"')
```

Comment:
I expect these args are invalid for Linux, will verify during the next phase of investigation